components/brave_wallet/browser/fil_transaction.cc

```diff
@@ -0,0 +1,177 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/fil_transaction.h"
+
+#include <algorithm>
+#include <string>
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/values.h"
+#include "brave/components/brave_wallet/common/fil_address.h"
+
+namespace brave_wallet {
+
+namespace {
+
+bool IsNumericString(const std::string& value) {
+  return std::all_of(value.begin(), value.end(),
+                     [](char c) { return std::isdigit(c) != 0; });
+}
+
+}  // namespace
+
+FilTransaction::FilTransaction() : gas_limit_(0) {}
+
+FilTransaction::FilTransaction(const FilTransaction&) = default;
+FilTransaction::FilTransaction(absl::optional<uint64_t> nonce,
+                               const std::string& gas_premium,
+                               const std::string& gas_fee_cap,
+                               int64_t gas_limit,
+                               const std::string& max_fee,
+                               const FilAddress& to,
+                               const std::string& value,
+                               const std::string& cid)
+    : nonce_(nonce),
+      gas_premium_(gas_premium),
+      gas_fee_cap_(gas_fee_cap),
+      gas_limit_(gas_limit),
+      max_fee_(max_fee),
+      cid_(cid),
+      to_(to),
+      value_(value) {}
+
+FilTransaction::~FilTransaction() = default;
+
+bool FilTransaction::IsEqual(const FilTransaction& tx) const {
+  return nonce_ == tx.nonce_ && gas_premium_ == tx.gas_premium_ &&
+         gas_fee_cap_ == tx.gas_fee_cap_ && gas_limit_ == tx.gas_limit_ &&
+         to_ == tx.to_ && value_ == tx.value_ && cid_ == tx.cid_;
```

Comment:
does `max_fee` not need to be compared here?

Reply:
sure, thanks, added